adrs/3122196229/README.md

```diff
@@ -0,0 +1,55 @@
+# NodeJS files and directories names conversion
+
+* **State:** Draft
+* **Created:** 2020-11-10
+* **Tags:** nodejs
+
+## Context
+
+Across the ecosystem, we can find all sort of files and directories naming
+conversions in NodeJS projects.
+
+That has caused misalignment, shifting the focus away from more important
+matters, also have created conventions that introduce problems such:
+
+* Disagreement around English language, making it even more difficult for the
+  non-English speaker to figure out the right decision.
+* Case sensitive vs. case insensitive file systems, causing issues in different
+  environments such as your local computer vs. CI/CD computers.
+* Format mismatching around separating English words.
+
+The `camelCase`, and/or `PascalCase` for naming convention, although this seems
+a good solution there are some caveats.
+
+What happen for the following cases `httpClient` vs. `HTTPClient` vs.
+`HttpClient`?
+
+* Which one is the more appropriate choice here without getting into
+  disagreement and use the phrase "it depends"?
+
+In some cases, the file `PascalCase` is used if the file is a class or
+a React component, otherwise follow `camelCase` is used.
+
+* What happen when the file contains more than a class or a React component? How
+  do we take the appropriate decision about it without getting into
+  disagreement and use the phrase "it depends"?
+
+Also, does that means that if we refactor the code, and we move the React
+component out of a file, that file name must change?
+
+That would introduce more potential refactoring like renaming file paths,
+adding more unnecessary line of code changes.
+
+As well as issues in `CI` environments, where it may be the case that `git`
+didn't rename the file due to case-insensitive file systems, and you must use
+`git mv [old file] [new file]` to fix such issue.
+
+## Resolution
+
+File or directory names:
+
+* Should not contain any leading or trailing spaces
+* Must be lowercase i.e., no uppercase or mixed case names are allowed
```

Comment:
Do you have an example where this is considered best practice in **any** javascript project?

Outside of the argument of It's bEsT PrAcTicE, I believe camelCase is objectively easier to read when it comes to functions/classes/components that are comprised of multiple words.

Reply:
Sure, NodeJS itself https://github.com/nodejs/node/tree/master/lib

The only caveat is that NodeJS has some files starting with `_` which is illegal in NPM packages for that very reason if I am not mistaken.

And NodeJS uses `_` which is not allow in NPM package names so that is why it is dropped from the naming.

> Do you have an example where this is considered best practice in any javascript project?

Also, it was never a good practice to use camelCase or PascalCase until the React team made it a thing thou, ignoring all the issues around it.

Reply:
The best thing about "standards" is there are so many to choose from :)

Node.js uses `snake_case` for directory and file names. https://github.com/nodejs/node
So does Deno. https://github.com/denoland/deno

Next.js is one of the most starred Javascript/Typescript repos on github and it uses `param-case` for directory and file names. https://github.com/vercel/next.js
NPM too. https://github.com/npm/cli

To be clear, I agree with `camelCase` for function/variable names in JS and `ClassCase` for classes but that is for a different ADR. Also, those examples are cherry picked. They show that very popular/important repos do use these conventions but you can also find ones that don't. TypeScript for example https://github.com/microsoft/TypeScript

Reply:
I would like to suggest using `_` but that is not an allowed NPM package name, that is my only push back on that. Thoughts?

Either `-` and `_` do not require URL encoding so that is good.

Reply:
Right, that is the main reason, in cases, we move the directory into an `npm` package or whatever, we don't need to do anything special.